### PR TITLE
Dry up Snapshot ITs further

### PR DESCRIPTION
**Pull request by original-brownbear**
Some more obvious cleaning up of the snapshot ITs.

follow up to #58818

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/discovery/SnapshotDisruptionIT.java [hunk 1]
```diff
@@ -298,8 +294,7 @@ private void assertSnapshotExists(String repository, String snapshot) {
     }
 
     private void createRandomIndex(String idxName) throws InterruptedException {
-        assertAcked(prepareCreate(idxName, 0, Settings.builder().put("number_of_shards", between(1, 20))
-            .put("number_of_replicas", 0)));
+        assertAcked(prepareCreate(idxName, 0, indexSettingsZeroReplicas(between(1, 20))));
```
Comment (tlrx) [inline server/src/internalClusterTest/java/org/elasticsearch/discovery/SnapshotDisruptionIT.java:297]: Maybe we could lower this down?

File: server/src/test/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java [hunk 2]
```diff
@@ -322,4 +340,30 @@ protected SnapshotInfo createFullSnapshot(String repoName, String snapshotName)
...
         return snapshotInfo;
     }
+
+    protected void createIndexWithSomeData(String indexName, int docCount) throws InterruptedException {
+        createIndex(indexName);
+        ensureGreen();
+        indexSomeData(indexName, docCount);
+    }
+
+    protected void indexSomeData(String index, int numdocs) throws InterruptedException {
```
Comment (tlrx) [inline server/src/test/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java:350]: nit: I'd call this `indexRandomDocs()`

File: server/src/test/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java [hunk 3]
```diff
@@ -322,4 +340,30 @@ protected SnapshotInfo createFullSnapshot(String repoName, String snapshotName)
         assertThat(snapshotInfo.state(), is(SnapshotState.SUCCESS));
         return snapshotInfo;
     }
+
+    protected void createIndexWithSomeData(String indexName, int docCount) throws InterruptedException {
```
Comment (tlrx) [inline server/src/test/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java:344]: nit: I'd call this `createIndexWithRandomDocs()`

File: server/src/test/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java [hunk 4]
```diff
@@ -274,6 +279,19 @@ protected void createRepository(String repoName, String type, Path location) {
...
 
+    protected void createRepository(String repoName, String type) {
+        Settings.Builder settings = Settings.builder().put("location", randomRepoPath()).put("compress", randomBoolean());
+        if (rarely()) {
+            settings = settings.put("chunk_size", randomIntBetween(100, 1000), ByteSizeUnit.BYTES);
+        }
+        createRepository(repoName, type, settings);
+    }
+
+    protected static Settings.Builder indexSettingsZeroReplicas(int shards) {
```
Comment (tlrx) [inline server/src/test/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java:290]: I'd call this `indexSettingsNoReplicas()`

File: server/src/test/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java [hunk 5]
```diff
@@ -274,6 +279,19 @@ protected void createRepository(String repoName, String type, Path location) {
         createRepository(repoName, type, Settings.builder().put("location", location));
     }
 
+    protected void createRepository(String repoName, String type) {
+        Settings.Builder settings = Settings.builder().put("location", randomRepoPath()).put("compress", randomBoolean());
+        if (rarely()) {
```
Comment (tlrx) [inline server/src/test/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java:284]: +1

**Review by tlrx — APPROVED** [review]
LGTM - I left some naming suggestions, feel free to apply or not.

File: server/src/internalClusterTest/java/org/elasticsearch/discovery/SnapshotDisruptionIT.java (hunk 1 above)
Comment (original-brownbear) [reply server/src/internalClusterTest/java/org/elasticsearch/discovery/SnapshotDisruptionIT.java:297]: Yea definitely, moved it to `5` :)

**Comment by original-brownbear** [comment]
Thanks Tanguy!

**Pull request merged**